### PR TITLE
Add code coverage reporting to codecov

### LINKED ISSUE #52: Add CodeCov support to .travis.yml
Issue Summary

Add this to our travis.ci configuration so that code coverage tests will run!

https://github.com/codecov/example-ruby/blob/master/.travis.yml

### PR DESCRIPTION
**Pull request by rahulpuroht**
# Fixes # 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [ ] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- This PR add code coverage reporting and integrates it with codecov.
- 
Closes #52
If you have questions, please send an email to [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=54) <br/>All committers have signed the CLA.

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+  global:
+    - CC_TEST_REPORTER_ID=$CC_TEST_REPORTER_ID
 language: ruby
 rvm:
   - 2.2
+before_install:
+- gem install simplecov  
 before_script:
   - bundle install
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
```
Comment (lauratpa) [inline .travis.yml:11]: These changes to `.travis.yml` seem to be about adding code coverage support for CodeClimate service. 

If you want to add code coverage for codecov service you this file to reflect the setup described here: https://github.com/codecov/example-ruby

File: test/test_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+if ENV['CI'] == 'true'
+  require 'simplecov'
+  SimpleCov.start
+
+  require 'codecov'
```
Comment (lauratpa) [inline test/test_helper.rb:5]: In order to be able to require the codecov library you should install this gem before as well.

**Comment by thinkingserious** [comment]
Hello @rahulpuroht,

Could you please respond to the code review so that we can move this PR towards merging? Thanks!

With Best Regards,

Elmer

**Comment by childish-sambino** [comment]
Fixed by https://github.com/sendgrid/smtpapi-ruby/pull/87

**Pull request closed without merging**